### PR TITLE
Add test for irq_router

### PR DESCRIPTION
**Pull request by bluewww**
Adds basic test for configuring the `irq_router`

### CODE REVIEW

File: sw/include/regs/irq_router.h [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+#define IRQ_ROUTER_PARAM_REG_WIDTH 32
+
+// Target selection bitmask control register
+#define IRQ_ROUTER_IRQ_TARGET_MASK_REG_OFFSET(i) (0x0 + 0x4 * i)
+
+#define IRQ_ROUTER_NUM_IRQ_SRCS 59
+#define IRQ_ROUTER_NUM_TARGETS 4
+
+enum car_irq_router_target {
+    IRQ_ROUTER_TARGET_NONE          = 0,
```
Comment (alex96295) [inline sw/include/regs/irq_router.h:24]: I think the LSB (position 0) is the PLIC (shared among the cores)

Comment (alex96295) [reply sw/include/regs/irq_router.h:24]: Out of the 32 possible targets, we have

|---|---|---| --- | ---|---|
| 31 | ... | 3 | 2 | 1 | 0 |
| None | ... | safety island | cva61 | cva60 | plic |

Comment (alex96295) [reply sw/include/regs/irq_router.h:24]: Ref

image

File: sw/tests/bare-metal/hostd/irq_router_rw.c [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+    // Init the HW
+    car_init_start();
+
+    // Init Uart
+    uint32_t rtc_freq   = *reg32(&__base_regs, CHESHIRE_RTC_FREQ_REG_OFFSET);
+    uint64_t reset_freq = clint_get_core_freq(rtc_freq, 2500);
+    uart_init(&__base_uart, reset_freq, 115200);
+
+    // after reset, all irqs should be routed to cva6 core 0
```
Comment (alex96295) [inline sw/tests/bare-metal/hostd/irq_router_rw.c:44]:
```
// after reset, all irqs should be routed to cva6 core 0 and core 1 using the PLIC
```

File: sw/include/regs/irq_router.h [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    IRQ_ROUTER_TARGET_CVA6_0        = 1,
```
Comment (alex96295) [inline sw/include/regs/irq_router.h:25]: I think this should be CLIC_CVA6_0 (same below) -> I know that the target is a hart, but for cva6 we made this fully flexible with Nils if you remember, so, for instance, CVA60 as a hart has two interrupt controllers (PLIC and CLIC), which transaltes in two targets from the viewpoint of the irq router

**Review by alex96295 — APPROVED** [review]
I left some comments because I think we are missing one target in the drivers, for the rest great job thanks a lot Rob!

**Comment by bluewww** [comment]
Ok I see what you mean . I was confused why I only counted 3 IRQ route targets but now it makes sense.

**Comment by alex96295** [comment]
> Ok I see what you mean . I was confused why I only counted 3 IRQ route targets but now it makes sense.

Yeah, if you could fix it, then we merge

**Review by alex96295 — APPROVED** [review]
LGTM

**Pull request merged**